pygfx/renderers/wgpu/engine/renderer.py

```diff
@@ -321,33 +322,61 @@ def blend_mode(self):
         """
         return self._blend_mode
 
+    @staticmethod
+    def _register_blend_mode(blender_class=None):
+        """Register a new blender for usage with rendering pipelines.
+
+        Note that Blender classes are highly experimental and their inteface
+        is expected to change rapidly from pygfx version 0.7.0 to
+        version 1.0.0.
+        The permenant existance of this function is not guaranteed.
+
+        Use carefully (i.e. at your own risk) as you help us
+        test and validate PyGFX's more advanced features.
+        """
+
+        def _register_blend_mode(blender_class):
+            name = blender_class.name
+            if name in WgpuRenderer._blenders_available:
+                warn(
+                    f"Blend mode '{name}' is already registered. "
+                    f"Overwritting {name} with {blender_class}.",
+                    stacklevel=2,
+                )
+            WgpuRenderer._blenders_available[name] = blender_class
+            return blender_class
+
+        if blender_class is None:
+            # Support decorator syntax
+            return _register_blend_mode
+        else:
+            # As well as calling syntax
+            _register_blend_mode(blender_class)
+
     @blend_mode.setter
     def blend_mode(self, value):
+        # Without importing our standard blender module, the
+        # blenders will not be registered and available.
+        # since they import the renderer module
+        # we cannot have this import at the top level otherwise it
+        # creates a circular import
+        # https://github.com/pygfx/pygfx/pull/966
+        from . import blender
+
         # Massage and check the input
         if value is None:
             value = "default"
         value = value.lower()
         if value == "default":
             value = "ordered2"
-        # Map string input to a class
-        m = {
-            "additive": blender_module.AdditiveFragmentBlender,
-            "opaque": blender_module.OpaqueFragmentBlender,
-            "dither": blender_module.DitherFragmentBlender,
-            "ordered1": blender_module.Ordered1FragmentBlender,
-            "ordered2": blender_module.Ordered2FragmentBlender,
-            "weighted": blender_module.WeightedFragmentBlender,
-            "weighted_depth": blender_module.WeightedDepthFragmentBlender,
-            "weighted_plus": blender_module.WeightedPlusFragmentBlender,
-        }
-        if value not in m:
-            raise ValueError(
-                f"Unknown blend_mode '{value}', use any of {set(m.keys())}"
-            )
+
+        b = self._blenders_available.get(value)
+        if blender is None:
```

Comment:
```suggestion
        blender = self._blenders_available.get(value, None)
        if blender is None:
```

Reply:
And the `b()`below should be `blender()`

Reply:
This clashes with `from . import blender` in ruff.

Reply:
ok i worked around it in a different way